### PR TITLE
Automate for OCP-27689 and OCP-27690

### PR DESCRIPTION
**Pull request by duanwei33**
pass log:
https://mastern-jenkins-csb-openshift-qe.apps.ocp4.prod.psi.redhat.com/job/Runner-v3-smoke/3104/console
https://mastern-jenkins-csb-openshift-qe.apps.ocp4.prod.psi.redhat.com/job/Runner-v3-smoke/3107/console

@chao007 @jhou1 Could you please take a look?

### CODE REVIEW

File: features/storage/csi_clone.feature [hunk 1]
```diff
@@ -41,4 +41,96 @@ Feature: CSI clone testing related feature
...
+    And I execute on the pod:
+      | sh | -c | sync -f /mnt/local/testfile |
+    Then the step should succeed
+
+    # Clone mypvc-ori with 3Gi size
+    Given I obtain test data file "storage/csi/pvc-clone.yaml"
+    When I create a dynamic pvc from "pvc-clone.yaml" replacing paths:
+      | ["metadata"]["name"]                         | mypvc-clone  |
+      | ["spec"]["storageClassName"]                 | standard-csi |
+      | ["spec"]["resources"]["requests"]["storage"] | 3Gi          |
```
Comment (jhou1) [inline features/storage/csi_clone.feature:77]: How about using a smaller size, like 2Gi, 1.5Gi? To save resources

**Comment by duanwei33** [comment]
@liangxia @pruan-rht Could you help take a look?

File: features/storage/csi_clone.feature [hunk 2]
```diff
@@ -41,4 +41,96 @@ Feature: CSI clone testing related feature
...
+      | ["spec"]["resources"]["requests"]["storage"] | 1Gi          |
+    Then the step should succeed
+    Given I obtain test data file "storage/misc/pod.yaml"
+    When I run oc create over "pod.yaml" replacing paths:
+      | ["metadata"]["name"]                                         | mypod-clone |
+      | ["spec"]["volumes"][0]["persistentVolumeClaim"]["claimName"] | mypvc-clone |
+      | ["spec"]["containers"][0]["volumeMounts"][0]["mountPath"]    | /mnt/local  |
+    Then the step should succeed
+    Given 30 seconds have passed
+    And the "mypvc-clone" PVC status is :pending
```
Comment (pruan-rht) [inline features/storage/csi_clone.feature:129]: @duanwei33 is this `Pending` state will persist or just a transient state?  Catching transient state is problematic.

File: features/storage/csi_clone.feature [hunk 3]
```diff
@@ -41,4 +41,96 @@ Feature: CSI clone testing related feature
...
+      | ["spec"]["storageClassName"]                 | standard-csi |
+      | ["spec"]["resources"]["requests"]["storage"] | 1Gi          |
+    Then the step should succeed
+    Given I obtain test data file "storage/misc/pod.yaml"
+    When I run oc create over "pod.yaml" replacing paths:
+      | ["metadata"]["name"]                                         | mypod-clone |
+      | ["spec"]["volumes"][0]["persistentVolumeClaim"]["claimName"] | mypvc-clone |
+      | ["spec"]["containers"][0]["volumeMounts"][0]["mountPath"]    | /mnt/local  |
+    Then the step should succeed
+    Given 30 seconds have passed
```
Comment (pruan-rht) [inline features/storage/csi_clone.feature:128]: @duanwei33 anyway not to hard-code sleep time?

Comment (duanwei33) [reply features/storage/csi_clone.feature:128]: @pruan-rht, thanks for the suggestion, do you mean the following change?

>     Given I wait up to 30 seconds for the steps to pass:
>     """
>     When I run the :describe client command with:
>       | resource | pvc         |
>       | name     | mypvc-clone |
>     Then the step should succeed
>     And the output should match:
>       | ProvisioningFailed                                                                     |
>       | new PVC request must be greater than or equal in size to the specified PVC data source |
>     """

Comment (pruan-rht) [reply features/storage/csi_clone.feature:128]: @duanwei33 yes...

File: features/storage/csi_clone.feature [hunk 4]
```diff
@@ -41,4 +41,96 @@ Feature: CSI clone testing related feature
...
+      | ["metadata"]["name"]                                         | mypod-clone |
+      | ["spec"]["volumes"][0]["persistentVolumeClaim"]["claimName"] | mypvc-clone |
+      | ["spec"]["containers"][0]["volumeMounts"][0]["mountPath"]    | /mnt/local  |
+    Then the step should succeed
+    Given the pod named "mypod-clone" becomes ready
+    And the "mypvc-clone" PVC becomes :bound 
+    Given the expression should be true> pv(pvc("mypvc-clone").volume_name).capacity_raw(cached: false) == "3Gi"
+    When I execute on the "mypod-clone" pod:
+      | sh | -c | ls /mnt/local/testfile |
+    Then the step should succeed
```
Comment (liangxia) [inline features/storage/csi_clone.feature:90]: No need the `ls` check here.  As if you can use below `more` command to view the content, then it means the file exist, which is duplicate

File: features/storage/csi_clone.feature (hunk 1 above)
Comment (duanwei33) [reply features/storage/csi_clone.feature]: Updated to 2Gi

File: features/storage/csi_clone.feature (hunk 2 above)
Comment (duanwei33) [reply features/storage/csi_clone.feature]: Thanks, updated to use the waiting loop.

File: features/storage/csi_clone.feature (hunk 3 above)
Comment (duanwei33) [reply features/storage/csi_clone.feature]: Thanks, Updated.

File: features/storage/csi_clone.feature (hunk 4 above)
Comment (duanwei33) [reply features/storage/csi_clone.feature]: Thanks, removed.

**Comment by duanwei33** [comment]
@liangxia @pruan-rht 
Updated according to your comments, could you help review?
here is the [new pass log](https://privatebin-it-iso.int.open.paas.redhat.com/?b422d3ec68eacdc4#9QjnuBY2njKsTMU5md4u5Yb9fMsSHBnGPtY4FR4QmTWe)

**Pull request merged**